5.5/contrib/common.sh

```diff
@@ -0,0 +1,155 @@
+# https://dev.mysql.com/doc/refman/5.5/en/identifiers.html
+mysql_identifier_regex='^[a-zA-Z0-9_]+$'
+mysql_password_regex='^[a-zA-Z0-9_~!@#$%^&*()-=<>,.?;:|]+$'
+
+function usage() {
+  [ $# == 2 ] && echo "error: $1"
+  echo "You must specify following environment variables:"
+  echo "  MYSQL_USER (regex: '$mysql_identifier_regex')"
+  echo "  MYSQL_PASSWORD (regex: '$mysql_password_regex')"
+  echo "  MYSQL_DATABASE (regex: '$mysql_identifier_regex')"
+  echo "Optional:"
+  echo "  MYSQL_ROOT_PASSWORD (regex: '$mysql_password_regex')"
+  echo "Settings:"
+  echo "  MYSQL_LOWER_CASE_TABLE_NAMES (default: 0)"
+  echo "  MYSQL_MAX_CONNECTIONS (default: 151)"
+  echo "  MYSQL_FT_MIN_WORD_LEN (default: 4)"
+  echo "  MYSQL_FT_MAX_WORD_LEN (default: 20)"
+  echo "  MYSQL_AIO (default: 1)"
+  exit 1
+}
+
+function validate_variables() {
+  if ! [[ -v MYSQL_USER && -v MYSQL_PASSWORD && -v MYSQL_DATABASE ]]; then
+    usage
+  fi
+
+  [[ "$MYSQL_USER"     =~ $mysql_identifier_regex ]] || usage "Invalid MySQL username"
+  [ ${#MYSQL_USER} -le 16 ] || usage "MySQL username too long (maximum 16 characters)"
+  [[ "$MYSQL_PASSWORD" =~ $mysql_password_regex   ]] || usage "Invalid password"
+  [[ "$MYSQL_DATABASE" =~ $mysql_identifier_regex ]] || usage "Invalid database name"
+  [ ${#MYSQL_DATABASE} -le 64 ] || usage "Database name too long (maximum 64 characters)"
+  if [ -v MYSQL_ROOT_PASSWORD ]; then
+    [[ "$MYSQL_ROOT_PASSWORD" =~ $mysql_password_regex ]] || usage "Invalid root password"
+  fi
+}
+
+# Make sure env variables don't propagate to mysqld process.
+function unset_env_vars() {
+  unset MYSQL_USER MYSQL_PASSWORD MYSQL_DATABASE MYSQL_ROOT_PASSWORD
+}
+
+# Poll until MySQL responds to our ping.
+function wait_for_mysql() {
+  pid=$1 ; shift
+
+  while [ true ]; do
+    if [ -d "/proc/$pid" ]; then
+      mysqladmin --socket=/tmp/mysql.sock ping &>/dev/null && return 0
+    else
+      return 1
+    fi
+    echo "Waiting for MySQL to start ..."
+    sleep 1
+  done
+}
+
+# Initialize the MySQL database (create user accounts and the initial database)
+function initialize_database() {
+  echo 'Running mysql_install_db ...'
+  mysql_install_db --datadir=$MYSQL_DATADIR
+
+  # Now start mysqld and add appropriate users.
+  echo 'Starting mysqld to create users ...'
+  /opt/rh/mysql55/root/usr/libexec/mysqld \
+    --defaults-file=$MYSQL_DEFAULTS_FILE \
+    --skip-networking --socket=/tmp/mysql.sock &
+
+  mysql_pid=$!
+  wait_for_mysql $mysql_pid
+
+  [ -v MYSQL_DISABLE_CREATE_DB ] && return
+
+  mysqladmin $admin_flags -f drop test
+  mysqladmin $admin_flags create "${MYSQL_DATABASE}"
+
+mysql $mysql_flags <<-EOSQL
+    CREATE USER '${MYSQL_USER}'@'%' IDENTIFIED BY '${MYSQL_PASSWORD}';
+    GRANT ALL ON \`${MYSQL_DATABASE}\`.* TO '${MYSQL_USER}'@'%' ;
+    FLUSH PRIVILEGES ;
+EOSQL
+
+  if [ -v MYSQL_ROOT_PASSWORD ]; then
+mysql $mysql_flags <<-EOSQL
+      GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}';
+EOSQL
```

Comment:
I think you can indent the mysql command itself, but the GRANT line should have no spaces in the beginning. Also, you can remove the - from <<-EOF since that causes preceding tabs (but not spaces) to be ignored.